core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -377,6 +380,66 @@ public void addSupportedLanguage(Locale locale) {
         getLocales().loadMissingBundles();
     }
 
+    /**
+     * Registers an instance of a class to be registered as an injectable dependency.<br>
+     * The command manager will attempt to inject all fields in a command class that are annotated with
+     * {@link co.aikar.commands.annotation.Dependency} with the provided instance.
+     *
+     * @param clazz the class the injector should look for when injecting
+     * @param instance the instance of the class that should be injected
+     * @throws IllegalStateException when there is already an instance for the provided class registered
+     */
+    public <T> void registerDependency(Class<? extends T> clazz, T instance){
+        registerDependency(clazz, clazz.getName(), instance);
+    }
+
+    /**
+     * Registers an instance of a class to be registered as an injectable dependency.<br>
+     * The command manager will attempt to inject all fields in a command class that are annotated with
+     * {@link co.aikar.commands.annotation.Dependency} with the provided instance.
+     *
+     * @param clazz the class the injector should look for when injecting
+     * @param key the key which needs to be present if that
+     * @param instance the instance of the class that should be injected
+     * @throws IllegalStateException when there is already an instance for the provided class registered
+     */
+    public <T> void registerDependency(Class<? extends T> clazz, String key, T instance){
+        if(dependencies.containsRow(clazz) && dependencies.containsColumn(key)){
+            throw new IllegalStateException("There is already an instance of " + clazz.getName() + " with the key " + key + " registered!");
+        }
+
+        dependencies.put(clazz, key, instance);
+    }
+
+    /**
+     * Attempts to inject instances of classes registered with {@link CommandManager#registerDependency(Class, Object)}
+     * into all fields that are marked with {@link Dependency}.
+     *
+     * @param baseCommand the instance which fields should be filled
+     */
+    void injectDependencies(BaseCommand baseCommand){
+        for (Field field : baseCommand.getClass().getDeclaredFields()) {
```

Comment:
If getDeclaredFields is the one that only includes its own fields, then switch to the other. we want to include parent class fields too, but I'm not at a java doc atm.

Reply:
when we don't use getDeclaredFields, we can't inject private fields. do we want to just iterate over both?